Can't built new linux kernels 6.3‑rc2 6.3‑rc3

### What happened?

Can't built new linux kernels 6.3‑rc2 6.3‑rc3
on ubuntu 22.04

commenting this line fixed problem
https://github.com/armbian/build/blob/main/lib/functions/compilation/kernel-debs.sh#L374


### How to reproduce?

Command you gave to the framework!

### Branch

main (main development branch)

### On which host OS are you observing this problem?

Other

### Relevant log output

```shell
[🔨]   dpkg-deb: building package 'linux-dtb-edge-sc8280xp' in '/home/misko/armbian12/.tmp/work-fde4e23a-8f57-46b3-9258-7b35bd5bd3f0/tmp.0aiROwfVad//linux-dtb-edge-sc8280xp_23.05.0-trunk--6.3-rc3-Se8d0-D2a58-P1bd3-C5e95Hfe66-B262d_arm64.deb'.
[🌱] Packaging linux-headers [ sc8280xp linux-sc8280xp-edge ]
[🐛] linux-headers packaging [ /home/misko/armbian12/.tmp/work-fde4e23a-8f57-46b3-9258-7b35bd5bd3f0/tmp.NY2MGFsMRH ]
[🔨]   '/home/misko/armbian12/.tmp/work-fde4e23a-8f57-46b3-9258-7b35bd5bd3f0/tmp.NY2MGFsMRH/lib/modules/6.3.0-rc3-sc8280xp/build' -> '/usr/src/linux-headers-6.3.0-rc3-sc8280xp'
[🔨]   '/home/misko/armbian12/cache/sources/linux-kernel-worktree/6.3__sc8280xp__arm64/.config' -> '/home/misko/armbian12/.tmp/work-fde4e23a-8f57-46b3-9258-7b35bd5bd3f0/tmp.NY2MGFsMRH/usr/src/linux-headers-6.3.0-rc3-sc8280xp/.config'
[🔨]     RMOBJ    tools/acpidbg/
[🔨]     RM       acpidbg
[🔨]     RMINC    include
[🔨]     RMOBJ    tools/acpidump/
[🔨]     RM       acpidump
[🔨]     RMINC    include
[🔨]     RMOBJ    tools/ec/
[🔨]     RM       ec
[🔨]     RMINC    include
[🔨]     RMOBJ    tools/pfrut/
[🔨]     RM       pfrut
[🔨]     RMINC    include
[🔨]     CLEAN   arm64
[🔨]     CLEAN   libapi
[🔨]     CLEAN   libbpf
[🔨]     CLEAN   libsubcmd
[🔨]     CLEAN   libsymbol
[🔨]     CLEAN   libperf
[🔨]     CLEAN   fixdep
[🔨]     CLEAN   feature-detect
[🔨]     CLEAN   python
[🔨]     CLEAN   bpf-skel
[🔨]     CLEAN   coresight
[🔨]     CLEAN   core-objs
[🔨]     CLEAN   core-progs
[🔨]     CLEAN   core-gen
[🔨]     CLEAN   Documentation
[🔨]   Makefile:32: compiler "gcc" does not support the ARMv8.5 MTE extension.
[🔨]   Makefile:33: test program "mte" will not be created.
[🔨]     CLEAN    
[🔨]   make[2]: *** No rule to make target 'clean'.  Stop.
[🔨]   make[1]: *** No rule to make target 'clean'.  Stop.
[🔨]   make: *** [Makefile:173: vm_clean] Error 2
[💥] Error 2 occurred in main shell
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/lib/functions/compilation/kernel-debs.sh
+++ b/lib/functions/compilation/kernel-debs.sh
@@ -372,6 +372,15 @@ function kernel_package_callback_linux_headers() {
 
 	# ${temp_file_list} is left at WORKDIR for later debugging, will be removed by WORKDIR cleanup trap
 
+	# Small detour: in v6.3-rc1, in commit https://github.com/torvalds/linux/commit/799fb82aa132fa3a3886b7872997a5a84e820062,
+	#               the tools/vm dir was renamed to tools/mm. Unfortunately tools/Makefile still expects it to exist,
+	#               and "make clean" in the "/tools" dir fails. Drop in a fake Makefile there to work around this.
+	if [[ ! -f "${headers_target_dir}/tools/vm/Makefile" ]]; then
+		display_alert "Creating fake tools/vm/Makefile" "6.3+ hackfix" "warn"
+		mkdir -p "${headers_target_dir}/tools/vm"
+		echo -e "clean:\n\techo fake clean for tools/vm" > "${headers_target_dir}/tools/vm/Makefile"
+	fi
+
 	# Now, make the script dirs clean.
 	# This is run in our _target_ dir, NOT the source tree, so we're free to make clean as we wish without invalidating the next build's cache.
 	# Understand: I'm sending the logs of this to the bitbucket ON PURPOSE: "clean" tries to use clang, ALSA, etc, which are not available.

--- a/lib/functions/compilation/kernel-make.sh
+++ b/lib/functions/compilation/kernel-make.sh
@@ -53,8 +53,8 @@ function run_kernel_make_internal() {
 		"KGZIP=pigz" "KBZIP2=pbzip2" # Parallel compression, use explicit parallel compressors https://lore.kernel.org/lkml/20200901151002.988547791@linuxfoundation.org/ # @TODO: what about XZ?
 	)
 
-	# last statement, so it passes the result to calling function. "env -i" is used for empty env
-	full_command=("${KERNEL_MAKE_RUNNER:-run_host_command_logged}" "env" "-i" "${common_make_envs[@]}"
+	# last statement, so it passes the result to calling function. "env -i" is used for empty env. "pipetty" helps to get color logs.
+	full_command=("${KERNEL_MAKE_RUNNER:-run_host_command_logged}" "pipetty" "env" "-i" "${common_make_envs[@]}"
 		make "${common_make_params_quoted[@]@Q}" "$@" "${make_filter}")
 	"${full_command[@]}" # and exit with it's code, since it's the last statement
 }

